src/components/buttons/flat-button/flat-button.js

```diff
@@ -16,12 +16,19 @@ const getIconElement = props => {
   });
 };
 
-const getTextColor = (tone, isHover = false) => {
+const getTextColor = (tone, isHover = false, theme) => {
+  const overwrittenVars = {
+    ...vars,
+    ...theme,
+  };
+
   switch (tone) {
     case 'primary':
-      return isHover ? vars.colorPrimary25 : vars.colorPrimary;
+      return isHover
+        ? overwrittenVars.colorPrimary25
+        : overwrittenVars.colorPrimary;
     case 'secondary':
-      return vars.colorSolid;
+      return overwrittenVars.colorSolid;
```

Comment:
if it's a dark background, it makes more sense to wrap that background in a ThemeProvider, overriding the colorSolid.